solver/llbsolver/file/backend.go

```diff
@@ -251,6 +212,7 @@ func docopy(ctx context.Context, src, dest string, action pb.FileActionCopy, u *
 }
 
 type Backend struct {
+	Executor executor.Executor
```

Comment:
Wdyt if we put separate type definitions here for UNIX and windows? Logically the executor dependency really does not belong here, so would like to avoid it for UNIX case.

Another option would be to leave this type as-is and in `solve()` optionally pass in some option that does the user conversion. This would then only be passed in case of windows and linux can use the default implementation.

Reply:
> Logically the executor dependency really does not belong here

It felt a bit yucky to add the executor there. I changed it and added a new function `NewFileOpBackend()` that sets a `readUserFn` in the `Backend{}`. For Windows it wraps the `readUser()` function which uses the executor. For everything else we just return the usual `readUser()` function. This way we don't set an exported `Executor` field in the `Backend{}` struct.

Does that sound acceptable?